pkg/kubelet/volumes.go

```diff
@@ -143,7 +143,8 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 			return nil, err
 		}
 		if attacher != nil {
-			err = attacher.Attach(volSpec, kl.hostname)
+			host := &volumeHost{kl}
+			err = attacher.Attach(host, volSpec, kl.hostname)
```

Comment:
I think the challenge there is that the path of the device is determined by the call to AttachDisk not from the disk id as is the case with GCE. @rootfs is there a way around this ? I would say open a PR with the work done for AWS EBS and base that on another PR which has all the changes to the Attacher/Detacher interface that you need and let's carry on this discussion there. Otherwise we are trying to hit a moving target.